tests/regression/results.csv

```diff
@@ -2,18 +2,18 @@ Data,                               Config,                             Method,
 silesia.tar,                        level -5,                           compress simple,                    6738593
 silesia.tar,                        level -3,                           compress simple,                    6446372
 silesia.tar,                        level -1,                           compress simple,                    6186042
-silesia.tar,                        level 0,                            compress simple,                    4861423
+silesia.tar,                        level 0,                            compress simple,                    4861424
```

Comment:
Why does this change?

Reply:
Isn't level 0 also level 3, and therefore uses huffman?

Reply:
Indeed, `level 0 == level 3`, and `level 3` changes by the same amount.

Though it's somewhat unclear why this PR would introduce a (very minor) 1-byte regression on multiple samples.
The expectation is that it does impact huffman sorting speed only, but not the outcome of the sort operation,
which means the output should be identical ?

Reply:
Oh, I see now that the question was probably referring to the change itself rather than level 0 specifically.

The changes do affect the outcome of the sort operation. Because we now accept a larger range [0, 166] (previously [0, 114]) of symbols that get distinct bucketing, any symbols with counts in [115, 165] will not be sorted via quicksort, but directly inserted into buckets. So they can end up in a different ordering than before, which can affect the header generation.